### DISCUSSION TITLE
Use Webooks and Event rules to pass data

### DISCUSSION THREAD

**Discussion by Pulsar911n** [netbox-community/netbox#16166] [Help Wanted!]
Hey, I've been banging my head to my desk in the previous week because I can't figure out the following:
I recently got a task to automate host creation between netbox and zabbix. Whenever a host is created in netbox, the same host is supposed to be created in zabbix.

I've tried using a webhook to send a post request to the zabbix api endpoint, and triggering this webhook with an event rule. 

The problem is that I can't seem to find a way to give the hostname, ip, etc data to the request in the webhook.  Netbox stores these data in some way, but zabbix doesn't understand netbox's syntax, so im trying to figure out a way to pass data from netbox to zabbix in the correct format and method.

As far as I'm aware, the body template section of the webhook is responsible to sending data with the request. Said body template in my case looks like this: 
![image](https://github.com/netbox-community/netbox/assets/95418006/25e93b3f-8eb7-40e4-8cd9-f0b4f0a8a2d3)

I have tried to define the "host" parameter with netbox's syntax, data['name'], but this syntax doesn't match the rest of the request. thus it's not working.

Any ideas or help is appreciated.

**Comment by markkuleinio** [comment]
The template is Jinja2 templating, so this should work, based on the example in https://docs.netbox.dev/en/stable/integrations/webhooks/:

```
...
        "host": "{{ data['name'] }}",
...
```

**Reply by Pulsar911n** [reply]
I've tried what you suggested, but unfortunately nothing has changed. I'm starting to think that the problem might be with netbox calling the zabbix api endpoint. I managed to create a device using the api with a python script, but the same cannot be said to creating it with netbox.
